### PR TITLE
build.sh to build.cmd

### PR DESCRIPTION
**Pull request by ealib**
Convert and adapt the build script for Windows users.

Binaries are for `windows-amd64` and are built in `.\bin` as in the original script.

This PR does not require testing because it simply increases the tools available for binaries generation: the binaries themselves represent the passing of the test.

See this also as an interim step for #2275

### Pull Request Checklist

* [x] I have justified why this PR doesn't need tests.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/dendrite/blob/main/docs/CONTRIBUTING.md#sign-off)

Signed-off-by: `Emanuele Aliberti <dev@mtka.eu>`

### CODE REVIEW

File: build.cmd [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    set FLAGS=-X github.com/matrix-org/dendrite/internal.branch=%BRANCH% -X github.com/matrix-org/dendrite/internal.build=%BUILD%
+
+:CHECK_BIN
+    if exist "bin" goto :ALL_SET
+    mkdir "bin"
+
+:ALL_SET
+    set CGO_ENABLED=1
+    set GOOS=windows
+    set GOARCH=amd64
```
Comment (neilalexander) [inline build.cmd:39]: This seems like it will do the wrong thing on 32-bit Windows?

Comment (ealib) [reply build.cmd]: I removed hardwired `GOOS` and `GOARCH` to let the Go toolchain determine what the characteristics of the host are.

Comment (ealib) [reply build.cmd]: I updated `docs/INSTALL.md` too.

**Review by neilalexander — APPROVED** [review]
Thanks for this!

**Pull request merged**